### PR TITLE
add message to turn on/off indicator lights

### PR DESCRIPTION
**Pull request by mrivi**
Add message to turn on and off the vehicle lights:
- visible light
- IR lights
- position lights (green/red for orientation information)

### CODE REVIEW

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -1624,6 +1624,14 @@
         <description>Turns illuminators ON/OFF. An illuminator is a light source that is used for lighting up dark areas external to the sytstem: e.g. a torch or searchlight (as opposed to a light source for illuminating the system itself, e.g. an indicator light).</description>
         <param index="1" label="Enable" minValue="0" maxValue="1" increment="1">0: Illuminators OFF, 1: Illuminators ON</param>
       </entry>
+      <entry value="406" name="MAV_CMD_LIGHT_INDICATOR_ON_OFF" hasLocation="false" isDestination="false">
+        <wip/>
+        <!-- This message is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
+        <description>Turns light source ON/OFF.</description>
+        <param index="1" label="EnableEO" minValue="0" maxValue="1" increment="1">0: EO Indicator OFF, 1: EO Indicator ON</param>
```
Comment (dagar) [inline message_definitions/v1.0/common.xml:1631]: You can expand the labels if you want, they're just strings, not fields we can use programmatically.

```suggestion
        <param index="1" label="Electro-optical (EO)" minValue="0" maxValue="1" increment="1">0: EO Indicator OFF, 1: EO Indicator ON</param>
```

Comment (auturgy) [reply message_definitions/v1.0/common.xml:1631]: I'm not sure what the label is indicating.  Is it an illuminator for an EO camera?  Is it a visible spectrum navigation light / beacon?  Is it a status light on an EO camera?  It's just not clear.

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -1624,6 +1624,14 @@
...
+        <param index="2" label="EnableIR" minValue="0" maxValue="1" increment="1">0: IR Indicator OFF, 1: IR Indicator ON</param>
```
Comment (auturgy) [inline message_definitions/v1.0/common.xml:1632]: As per EO indicator.  Needs clarification

**Comment by auturgy** [comment]
Not sure if this is fine grained enough.  You'll likely want flash rates for these.

**Comment by mrivi** [comment]
@auturgy please have a look at the changes I made

File: message_definitions/v1.0/common.xml [hunk 3]
```diff
@@ -1624,6 +1624,16 @@
         <description>Turns illuminators ON/OFF. An illuminator is a light source that is used for lighting up dark areas external to the sytstem: e.g. a torch or searchlight (as opposed to a light source for illuminating the system itself, e.g. an indicator light).</description>
         <param index="1" label="Enable" minValue="0" maxValue="1" increment="1">0: Illuminators OFF, 1: Illuminators ON</param>
       </entry>
+      <entry value="406" name="MAV_CMD_LIGHT_INDICATOR_ON_OFF" hasLocation="false" isDestination="false">
+        <wip/>
+        <!-- This message is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
+        <description>Turns indicator light source ON/OFF. This light source is meant to show the presence of a vehicle.</description>
+        <param index="1" label="EnableVisibleLight" minValue="0" maxValue="1" increment="1">Visibile light 0: EO Indicator OFF, 1: EO Indicator ON</param>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:1631]: typo Visibile

File: message_definitions/v1.0/common.xml [hunk 4]
```diff
@@ -1624,6 +1624,16 @@
...
+        <param index="2" label="EnableInfraRed" minValue="0" maxValue="1" increment="1">0: IR Indicator OFF, 1: IR Indicator ON</param>
+        <param index="3" label="EnablePositionLight" minValue="0" maxValue="1" increment="1">Position lights (red/green) to indicate the vehicle heading, 0: Position lights OFF, 1: Positions Lights ON</param>
+        <param index="4" label="BlinkDuration" units="seconds">Portion of the period where the light is ON. </param>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:1634]: Do all the lights blink, and do they all blink at same rate and period? The way this message is set up I would assume that any light that is on will also blink at the same specified rate.

**Comment by hamishwillee** [comment]
Any synergy with [MAV_CMD_ILLUMINATOR_ON_OFF](https://mavlink.io/en/messages/common.html#MAV_CMD_ILLUMINATOR_ON_OFF)? I.e. could we merge one/supercede one?

**Comment by hamishwillee** [comment]
@auturgy This command useful for ArduPilot too? I.e. under new governance approach it can go into common.xml once their is an implementation?

File: message_definitions/v1.0/common.xml (hunk 4 above)
Comment (mrivi) [reply message_definitions/v1.0/common.xml:1634]: My intent was that you have only one enabled and the rate applied to the enabled one.

**Comment by mrivi** [comment]
@hamishwillee I have separated it from the illuminator because of the  different concept
- illuminator: gives light to an area such that the vehicle can see the environment with cameras etc
- indicator: the light source is meant to show the presence of the vehicle

File: message_definitions/v1.0/common.xml (hunk 4 above)
Comment (hamishwillee) [reply message_definitions/v1.0/common.xml:1634]: The thing is that there is nothing in the message that captures this intent or prevents it being mis-used.  To do what you want change param 1 to an enum with the different options you have specified (IR, visible, etc) and remove the specific params. Then it is only possible to enable one light at a time, and it is obvious how you are applying it. Also, if we come up with more light sources it is easy to extend. 

And now you can see why I thought that illuminator might be a reasonable option - this is really turn something on an off, possibly with a cycle.

**Comment by mrivi** [comment]
@hamishwillee I made the changes

**Comment by hamishwillee** [comment]
@mrivi 
- Thanks, this looks good to me. 
- FYI Mavlink moving towards a model where WIP means we have a working implementation a flight stack, so this is here for a shakedown couple of months while we do testing. Is this the case?

@auturgy Can you please re-review/are you OK with this going in (i.e. ArduPilot interested in implementing this).

**Pull request closed without merging**